src/BufferedFileReader.hpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file BufferedFileReader.hpp Code to read data from a file. The same compression algorithms as for the
+ * BufferedFileWriter are supported.
+ *
+ * This is part of the DUNE DAQ , copyright 2021.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+#ifndef READOUT_SRC_BUFFEREDFILEREADER_HPP_
+#define READOUT_SRC_BUFFEREDFILEREADER_HPP_
+
+#include "ReadoutIssues.hpp"
+#include "readout/ReadoutLogging.hpp"
+
+#include "logging/Logging.hpp"
+
+#include <boost/align/aligned_allocator.hpp>
+#include <boost/iostreams/stream.hpp>
+#include <boost/iostreams/stream_buffer.hpp>
+#include <boost/iostreams/device/file_descriptor.hpp>
+#include <boost/iostreams/filtering_stream.hpp>
+#include <boost/iostreams/filter/zstd.hpp>
+#include <boost/iostreams/filter/lzma.hpp>
+#include <boost/iostreams/filter/zlib.hpp>
+
+#include <iostream>
+#include <fstream>
+#include <limits>
+#include <string>
+#include <fcntl.h>
+#include <unistd.h>
+
+using namespace dunedaq::readout::logging;
+
+namespace dunedaq {
+  namespace readout {
+    /**
+     * Class to read data of a specified type in a buffered manner.
+     * @tparam RawType Type of the data that is read from the file.
+     * @tparam Alignment The alignment size of internal buffers.
+     */
+    template<class RawType, size_t Alignment=4096>
+    class BufferedFileReader {
+      using io_source_t = boost::iostreams::file_descriptor_source;
+      using aligned_allocator_t = boost::alignment::aligned_allocator<io_source_t::char_type, Alignment>;
+      using filtering_istream_t = boost::iostreams::filtering_stream<boost::iostreams::input, char, std::char_traits<char>, aligned_allocator_t>;
+    public:
+      /**
+       * Constructor to construct and initalize an instance. The file will be open after initialization.
+       * @param filename The file to be used.
+       * @param buffer_size The size of the buffer to be used.
+       * @param compression_algorithm The compression algorithm to use. Can be one of: None, zstd, lzma or zlib
+       * @throw CannotOpenFile If the file can not be opened.
+       * @throw ConfigurationError If the compression algorithm parameter is not recognized.
+       */
+      explicit BufferedFileReader(std::string filename, size_t buffer_size, std::string compression_algorithm = "None")
```

Comment:
Only single parameter constructors should be explicit. Please remove the specifier.